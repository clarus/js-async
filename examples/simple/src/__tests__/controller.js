// @flow
import 'babel-polyfill';
import {createStore} from 'redux';
import * as Ship from 'redux-ship';
import * as Effect from '../effect';
import * as Controller from '../controller';
import * as Model from '../model';

function runEffect(effect: Effect.Effect) {
  switch (effect.type) {
  case 'HttpRequest':
    if (effect.url === 'https://swapi.co/api/people/3/') {
      return JSON.stringify({
        films: [
          'https://swapi.co/api/films/2/', 
          'https://swapi.co/api/films/5/', 
          'https://swapi.co/api/films/4/', 
          'https://swapi.co/api/films/6/', 
          'https://swapi.co/api/films/3/', 
          'https://swapi.co/api/films/1/', 
          'https://swapi.co/api/films/7/'
        ]
      });
    }
    return JSON.stringify({title: `Title of ${effect.url}`});
  default:
    return;
  }
}

test('list the movies', async () => {
  const action = {type: 'Load'};
  const store = createStore(Model.reduce, Model.initialState);
  const ship = Ship.snap(Controller.control(action));
  const snapshot = await Ship.run(runEffect, store, ship);
  expect(snapshot).toMatchSnapshot();
});

