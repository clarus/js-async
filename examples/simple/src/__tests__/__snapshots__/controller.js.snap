// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list the movies 1`] = `
Object {
  "result": undefined,
  "snapshot": Array [
    Object {
      "state": null,
      "type": "GetState",
    },
    Object {
      "commit": Object {
        "type": "LoadStart",
      },
      "type": "Commit",
    },
    Object {
      "effect": Object {
        "ms": 500,
        "type": "Delay",
      },
      "type": "Effect",
    },
    Object {
      "effect": Object {
        "type": "HttpRequest",
        "url": "https://swapi.co/api/people/3/",
      },
      "result": "{\\"films\\":[\\"https://swapi.co/api/films/2/\\",\\"https://swapi.co/api/films/5/\\",\\"https://swapi.co/api/films/4/\\",\\"https://swapi.co/api/films/6/\\",\\"https://swapi.co/api/films/3/\\",\\"https://swapi.co/api/films/1/\\",\\"https://swapi.co/api/films/7/\\"]}",
      "type": "Effect",
    },
    Object {
      "snapshots": Array [
        Array [
          Object {
            "effect": Object {
              "type": "HttpRequest",
              "url": "https://swapi.co/api/films/2/",
            },
            "result": "{\\"title\\":\\"Title of https://swapi.co/api/films/2/\\"}",
            "type": "Effect",
          },
        ],
        Array [
          Object {
            "effect": Object {
              "type": "HttpRequest",
              "url": "https://swapi.co/api/films/5/",
            },
            "result": "{\\"title\\":\\"Title of https://swapi.co/api/films/5/\\"}",
            "type": "Effect",
          },
        ],
        Array [
          Object {
            "effect": Object {
              "type": "HttpRequest",
              "url": "https://swapi.co/api/films/4/",
            },
            "result": "{\\"title\\":\\"Title of https://swapi.co/api/films/4/\\"}",
            "type": "Effect",
          },
        ],
        Array [
          Object {
            "effect": Object {
              "type": "HttpRequest",
              "url": "https://swapi.co/api/films/6/",
            },
            "result": "{\\"title\\":\\"Title of https://swapi.co/api/films/6/\\"}",
            "type": "Effect",
          },
        ],
        Array [
          Object {
            "effect": Object {
              "type": "HttpRequest",
              "url": "https://swapi.co/api/films/3/",
            },
            "result": "{\\"title\\":\\"Title of https://swapi.co/api/films/3/\\"}",
            "type": "Effect",
          },
        ],
        Array [
          Object {
            "effect": Object {
              "type": "HttpRequest",
              "url": "https://swapi.co/api/films/1/",
            },
            "result": "{\\"title\\":\\"Title of https://swapi.co/api/films/1/\\"}",
            "type": "Effect",
          },
        ],
        Array [
          Object {
            "effect": Object {
              "type": "HttpRequest",
              "url": "https://swapi.co/api/films/7/",
            },
            "result": "{\\"title\\":\\"Title of https://swapi.co/api/films/7/\\"}",
            "type": "Effect",
          },
        ],
      ],
      "type": "All",
    },
    Object {
      "commit": Object {
        "movies": Array [
          "Title of https://swapi.co/api/films/2/",
          "Title of https://swapi.co/api/films/5/",
          "Title of https://swapi.co/api/films/4/",
          "Title of https://swapi.co/api/films/6/",
          "Title of https://swapi.co/api/films/3/",
          "Title of https://swapi.co/api/films/1/",
          "Title of https://swapi.co/api/films/7/",
        ],
        "type": "LoadSuccess",
      },
      "type": "Commit",
    },
  ],
}
`;
